subprojects/docs/src/docs/userguide/upgrading_version_4.adoc

```diff
@@ -328,6 +377,149 @@ To override this behavior, you can explicitly declare the configuration to use i
+ * The Google Cloud Storage JSON API Client Library used to access Google Cloud Storage backed Maven/Ivy repositories has been upgraded from v1-rev116-1.23.0 to v1-rev136-1.25.0.
+ * Ivy has been upgraded from 2.2.0 to http://ant.apache.org/ivy/history/2.3.0/release-notes.html[2.3.0].
+ * The JUnit Platform libraries used by the `Test` task have been upgraded from 1.0.3 to 1.3.1.
+ * The Maven Wagon libraries used to access Maven repositories have been upgraded from 2.4 to 3.0.0.
+ * SLF4J has been upgraded from 1.7.16 to https://www.slf4j.org/news.html[1.7.25].
+
+[[rel5.0:dependency_resolution]]
+=== [5.0] Fixes to dependency resolution
+
+Dependency resolutions fixes have been included in this release.
+By definition this could impact the set of resolved dependencies of your build.
+However the fixed issues are mostly about corner cases and combination with recent features and thus should have a limited impact.
+
+When a dependency constraint matched a real dependency, it was made part of the graph.
+However if for some reason the dependency was later evicted from the graph, the constraint remained present. Now, when the last non-constraint edge to a dependency disappears, all constraints for that dependency will be properly removed from the graph.
+
+[[rel5.0:changes_to_default_task]]
+=== [5.0] Changes to property factory methods on DefaultTask
+
+Property factory methods on DefaultTask are final
+The property factory methods such as newInputFile() are intended to be called from the constructor of a type that extends DefaultTask. These methods are now final to avoid subclasses overriding these methods and using state that is not initialized.
+
+Inputs and outputs are not automatically registered
+The Property instances that are returned by these methods are no longer automatically registered as inputs or outputs of the task. The Property instances need to be declared as inputs or outputs in the usual ways, such as attaching annotations such as @OutputFile or using the runtime API to register the property.
+
+Previously:
+
+[source, groovy]
+----
+class MyTask extends DefaultTask {
+    // note: no annotation here
+    final RegularFileProperty outputFile = newOutputFile()
+}
+
+task myOtherTask {
+    def outputFile = newOutputFile()
+    doLast { ... }
+}
+----
+
+Now:
+
+[source, groovy]
+----
+class MyTask extends DefaultTask {
+    @OutputFile // property needs an annotation
+    final RegularFileProperty outputFile = project.objects.fileProperty()
+}
+
+task myOtherTask {
+    def outputFile = project.objects.fileProperty()
+    outputs.file(outputFile) // or to be registered using the runtime API
+    doLast { ... }
+}
+----
+
+[[rel5.0:jaxb_and_java9]]
+=== [5.0] Gradle now bundles JAXB for Java 9 and above
+
+In order to use S3 backed artifact repositories, it was previously required to add `--add-modules java.xml.bind` to `org.gradle.jvmargs` when running on Java 9 and above.
```

Comment:
"S3-backed" (hyphen). Also, replace "it was previously require to add" with "you previously had to add...".